src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -239,12 +249,17 @@ private static ParameterDescriptor CreateParameterDescriptor(ParameterInfo param
             {
                 return (BindingSource.Body, parameter.Name ?? string.Empty, fromBodyAttribute.AllowEmpty, parameter.ParameterType);
             }
+            else if (attributes.OfType<IFromFileMetadata>().FirstOrDefault() is { } formFileAttribute)
+            {
+                return (BindingSource.FormFile, formFileAttribute.Name ?? parameter.Name ?? string.Empty, false, parameter.ParameterType);
+            }
             else if (parameter.CustomAttributes.Any(a => typeof(IFromServiceMetadata).IsAssignableFrom(a.AttributeType)) ||
                      parameter.ParameterType == typeof(HttpContext) ||
                      parameter.ParameterType == typeof(HttpRequest) ||
                      parameter.ParameterType == typeof(HttpResponse) ||
                      parameter.ParameterType == typeof(ClaimsPrincipal) ||
                      parameter.ParameterType == typeof(CancellationToken) ||
+                     parameter.ParameterType == typeof(IFormFileCollection) ||
```

Comment:
Why is this not also `BindingSource.FormFile`?

Reply:
So it'll fall into this block as it binds to a `FormFile`, rather than `Services`.

https://github.com/dotnet/aspnetcore/blob/05cbaa0f4bb99a8d4f54daba4fb7e85483d324aa/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L283-L286

Reply:
Ah, sorry, I see what you mean now.

Yeah I wasn't sure if this counted as `Form` (because that's where it ultimately comes from), `FormFile` (even though there might be multiple files, so then is it being _a_ form file correct?) or `Services`.

I went with `Services` to start with, but I can change it to one of the other two.